lib/quiz/match_the_following.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:flutter/material.dart';
+
+class MatchingGame extends StatefulWidget {
+  Map<String, dynamic> gameData;
+  MatchingGame(
+      {key,
+      this.gameData = const {
+        "image":
+            "https://raw.githubusercontent.com/flutter/website/master/_includes/code/layout/lakes/images/lake.jpg",
+        "question": "what is this game about?",
+        "pairs": {
+          "1": "string1",
+          "2": "string2",
+          "3": "string3",
+          "4": "string4",
+          "5": "string5",
+          "6": "string6",
+          "7": "string7",
+          "8": "string8",
+          "9": "string9",
+          "10": "string10",
+          "11": "string11",
+          "12": "string12",
+          "13": "string13",
+          "14": "string14",
+          "15": "string15",
+          "16": "string16",
+          "17": "string17",
+          "18": "string18",
+          "19": "string19",
+          "20": "string20"
+        }
+      }})
+      : super(key: key);
+  @override
+  _MatchingGameState createState() => new _MatchingGameState();
+}
+
+class _MatchingGameState extends State<MatchingGame> {
+  Map<String, dynamic> _selected = {};
+  String _leftItemTapped;
+
+  List<String> _leftSideItems = [];
+  List<String> _leftSideDisabledItems = [];
+  List<dynamic> _rightSideDisabledItems = [];
+  List<dynamic> _rightSideItems = [];
+
+  @override
+  void initState() {
+    super.initState();
+    _leftSideItems = (widget.gameData["pairs"].keys.toList()..shuffle());
+    _rightSideItems = (widget.gameData["pairs"].values.toList()..shuffle());
+  }
+
+  bool _checkItem(String i, bool sideCheck) {
+    if (sideCheck == true) {
```

Comment:
Name of the variable is not meaningful. Since you are checking whether it is left or right, it should be leftSideCheck or isLeft or something like that

Reply:
Corrected both the names of the variable.